axelrod/strategies/finite_state_machines.py

```diff
@@ -21,23 +22,48 @@ def __init__(self, transitions, initial_state):
         with initial play C and initial state 1.
 
         """
-        self.state = initial_state
-        self.state_transitions = dict()
-        for (state, opp_action, next_state, next_action) in transitions:
-            self.state_transitions[(state, opp_action)] = (next_state, next_action)
+        self._state = initial_state
+        self._state_transitions = {(current_state, input_action): (next_state, output_action) for
+                                   current_state, input_action, next_state, output_action in transitions}  # type: dict
+
+        self._raise_error_for_bad_input()
+
+    def _raise_error_for_bad_input(self):
+        callable_states = [self._state] + [pair[0] for pair in self._state_transitions.values()]
+        for state in callable_states:
+            self._raise_error_for_bad_state(state)
+
+    def _raise_error_for_bad_state(self, state: int):
+        if (state, C) not in self._state_transitions or (state, D) not in self._state_transitions:
+            raise ValueError('state: {} does not have values for both C and D'.format(state))
+
+    @property
+    def state(self) -> int:
+        return self._state
+
+    @state.setter
+    def state(self, new_state: int):
+        self._raise_error_for_bad_state(new_state)
+        self._state = new_state
+
+    @property
+    def state_transitions(self) -> dict:
+        return self._state_transitions.copy()
 
     def move(self, opponent_action: Action) -> Action:
         """Computes the response move and changes state."""
-        next_state, next_action = self.state_transitions[(self.state, opponent_action)]
-        self.state = next_state
+        next_state, next_action = self._state_transitions[(self._state, opponent_action)]
+        self._state = next_state
         return next_action
 
-    def __eq__(self, other: Player) -> bool:
+    def __eq__(self, other) -> bool:
         """Equality of two FSMs"""
-        check = True
-        for attr in ["state", "state_transitions"]:
-            check = check and getattr(self, attr) == getattr(other, attr)
-        return check
+        if not isinstance(other, SimpleFSM):
+            return False
+        return (self._state, self._state_transitions) == (other.state, other.state_transitions)
+
+    def __ne__(self, other):
```

Comment:
I believe it's no longer necessary in py3, can we remove it (less lines of code to maintain :)) and just have a test that checks we have the `__ne__` behaviour?